ci/upload_artifactbundle.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+# This script generates an artifactbundle for protoc. This artifactbundle
+# is used by the Swift package manger. The script is run by a GitHub action
+# when a new release is created.
+
+set -ex
+
+# We have to filter for the correct release. protoc releases are following the vX.X scheme
+# but the repo also has language specific vX.X.X releases. $GITHUB_REF is in the format of refs/tags/<tag_name>
+TAG=$GITHUB_REF
+TAG="${TAG#???????????}"
+
+if [[ ! "$TAG" =~ ^[0-9]+\.[0-9]+$ ]]; then
+    echo "Error: $TAG does not match the expected pattern"
+    exit 1
+fi
+
+AUTH="Authorization: token $GITHUB_TOKEN"
+
+# Fetch all protoc release assets
+curl -LJ --output protoc-$TAG-osx-x86_64.zip -H 'Accept: application/octet-stream' https://github.com/$GITHUB_REPOSITORY/releases/download/v$TAG/protoc-24.3-osx-x86_64.zip
+curl -LJ --output protoc-$TAG-osx-aarch_64.zip -H 'Accept: application/octet-stream' https://github.com/$GITHUB_REPOSITORY/releases/download/v$TAG/protoc-24.3-osx-aarch_64.zip
+curl -LJ --output protoc-$TAG-linux-aarch_64.zip -H 'Accept: application/octet-stream' https://github.com/$GITHUB_REPOSITORY/releases/download/v$TAG/protoc-24.3-linux-aarch_64.zip
+curl -LJ --output protoc-$TAG-linux-x86_64.zip -H 'Accept: application/octet-stream' https://github.com/$GITHUB_REPOSITORY/releases/download/v$TAG/protoc-24.3-linux-x86_64.zip
+curl -LJ --output protoc-$TAG-win64.zip -H 'Accept: application/octet-stream' https://github.com/$GITHUB_REPOSITORY/releases/download/v$TAG/protoc-24.3-win64.zip
+
+# Unzip all assets
+mkdir protoc-$TAG.artifactbundle
+unzip -d protoc-$TAG.artifactbundle/protoc-$TAG-osx-x86_64 protoc-$TAG-osx-x86_64.zip
+unzip -d protoc-$TAG.artifactbundle/protoc-$TAG-osx-aarch_64 protoc-$TAG-osx-aarch_64.zip
+unzip -d protoc-$TAG.artifactbundle/protoc-$TAG-linux-aarch_64 protoc-$TAG-linux-aarch_64.zip
+unzip -d protoc-$TAG.artifactbundle/protoc-$TAG-linux-x86_64 protoc-$TAG-linux-x86_64.zip
+unzip -d protoc-$TAG.artifactbundle/protoc-$TAG-win64 protoc-$TAG-win64.zip
+
+# Create info.json for artifactbundle
+cat > protoc-$TAG.artifactbundle/info.json << EOF
```

Comment:
Looks like this essentially just re-zips our existing github release protoc artifacts into a different structure and adds a info.json file.

Is this really necessary for SwiftPM use? Could SwiftPM users use existing protoc zips e.g. if we included corresponding info.json files?

We would prefer to minimize support burden here esp. once we support building protoc in SwiftPM in the future.

Reply:
At the moment, this is the only possible way to get SwiftPM to consume `protoc`. We have to zip up the existing zips and add the JSON file. There is no way to use the existing zips and add JSON files to them because we need to bundle multiple different architectures into a single zip. So there is always the need for at least one new asset.

As an investigative task I also did some work to build `protoc` via SwiftPM and got that working. There are two downsides for this:
1. We need to maintain another build system here that can build `protoc`
2. Building protoc from scratch is adding additional build time.